tests/test_default_qubit.py

```diff
@@ -854,6 +929,70 @@ def circuit(x):
 
         assert np.isclose(np.mean(runs), -np.sin(p), atol=tol, rtol=0)
 
+    @pytest.mark.parametrize("name,expected_output", [
+        ("PauliX", 1),
+        ("PauliY", 1),
+        ("S", -1),
+    ])
+    def test_inverse_circuit(self, qubit_device_1_wire, tol, name, expected_output):
+        """Tests the inverse of supported gates that act on a single wire and are not parameterized"""
+
+        op = getattr(qml.ops, name)
+
+        @qml.qnode(qubit_device_1_wire)
+        def circuit():
+            qml.BasisState(np.array([1]), wires=[0])
+            op(wires=0).inv()
+            return qml.expval(qml.PauliZ(0))
+
+        assert np.isclose(circuit(), expected_output, atol=tol, rtol=0)
+
+    @pytest.mark.parametrize("name,expected_output", [
+        ("PauliX", 1),
+        ("PauliY", 1),
+        ("S", -1),
+    ])
+    def test_inverse_circuit_calling_inv_multiple_times(self, qubit_device_1_wire, tol, name, expected_output):
+        """Tests that multiple calls to the inverse of an operation works"""
+
+        op = getattr(qml.ops, name)
+
+        @qml.qnode(qubit_device_1_wire)
+        def circuit():
+            qml.BasisState(np.array([1]), wires=[0])
+            op(wires=0).inv().inv().inv()
+            return qml.expval(qml.PauliZ(0))
+
+        assert np.isclose(circuit(), expected_output, atol=tol, rtol=0)
+
+    @pytest.mark.parametrize("name,expected_output,phi", [("RX", 1,
+                                                           multiplier * 0.5432) for multiplier in range(25)
+                                                          ])
+    def test_inverse_circuit_with_parameters(self, qubit_device_1_wire, tol, name, expected_output, phi):
+        """Tests the inverse of supported gates that act on a single wire and are parameterized"""
+
+        @qml.qnode(qubit_device_1_wire)
+        def circuit():
+            qml.RX(phi, wires=0)
+            qml.RX(phi, wires=0).inv()
+            return qml.expval(qml.PauliZ(0))
+
+        assert np.isclose(circuit(), expected_output, atol=tol, rtol=0)
+
+
+
+    @pytest.mark.parametrize("name,expected_output,phi", [("RX", 1,
+                                                           multiplier * 0.5432) for multiplier in range(25)
```

Comment:
:laughing: 25 is probably excessive. I think we can get away with about 5-8 cases